tests/testthat/test-resp-url.R

```diff
@@ -0,0 +1,7 @@
+test_that("can extract url components from a response", {
+  resp <- req_perform(request_test("/get?a=1&b=2"))
+
+  expect_equal(resp_url(resp), paste0(example_url(), "get?a=1&b=2"))
+  expect_equal(resp_url_path(resp), "/get")
+  expect_equal(resp_url_query(resp), list(a = "1", b = "2"))
```

Comment:
Probably not really worth the complexity but this examples makes me wonder whether something like a `ptypes` argument would make sense. It would be nice to easily get `list(a = 1, b = 2)`.